src/node/gov.h

```diff
@@ -12,33 +12,53 @@ namespace ccf
   namespace jsgov
   {
     using ProposalId = std::string;
+    using Ballots = std::unordered_map<ccf::MemberId, std::string>;
+    using Votes = std::unordered_map<ccf::MemberId, bool>;
+
+    struct Failure
+    {
+      std::string reason;
+      std::string trace;
+      bool operator==(const Failure& rhs) const
+      {
+        return reason == rhs.reason && trace == rhs.trace;
+      }
+    };
+    DECLARE_JSON_TYPE(Failure);
+    DECLARE_JSON_REQUIRED_FIELDS(Failure, reason, trace);
```

Comment:
Minor: Since `trace` is sometimes/often the empty string (if `js_error_message` doesn't return one), I think it should be an optional field here so we don't insert unnecessary empty string fields in the JSON serialisation.

Reply:
Yes, good point.